indico/modules/events/abstracts/models/abstracts.py

```diff
@@ -571,6 +574,26 @@ def get_track_reviewing_state(self, track):
         else:
             return AbstractReviewingState.mixed
 
+    def get_avg_per_question_per_track(self):
+        ratings = {}
+        abstract_ratings = (AbstractReviewRating.query
+                            .join(AbstractReviewRating.review)
+                            .join(AbstractReviewRating.question)
+                            .options(contains_eager('review'), contains_eager('question'))
+                            .filter(AbstractReview.abstract == self,
+                                    ~AbstractReviewQuestion.is_deleted)
+                            .all())
+        for rating in abstract_ratings:
+            if rating.question.no_score:
+                continue
+            ratings.setdefault(rating.review.track_id, {})
```

Comment:
This would replace the `ratings = {}`. As soon as you access a ratings[foo] it'll add that key with a new dict as the value. Which is exactly what you do here..